### PR TITLE
fix #271198: Custom noteheads implementation for 2.3

### PR DESCRIPTION
**Pull request by anatoly-os**
Based on #3631 + additional font fallback implementation.

### CODE REVIEW

File: fonts/mscore/glyphnames.json [hunk 1]
```diff
@@ -745,5 +743,81 @@
     },
     "noteLongaSquareUp": {
         "codepoint": "U+e1c4"
+    },
```
Comment (lasconic) [inline fonts/mscore/glyphnames.json]: I'm just starting, so maybe I miss something but if the font doesn't have these glyphs, then they shouldn't be listed here.

File: mscore/editdrumset.cpp [hunk 2]
```diff
@@ -102,14 +118,83 @@ EditDrumset::EditDrumset(const Drumset* ds, QWidget* parent)
...
+            for (auto symName : (*MuseScore::bravuraRanges())[range]) {
+                   SymId id = Sym::name2id(symName);
+                   if (!excludeSym.contains(symName)) {
+                         QIcon icon;
+                         QPixmap image(w, h);
+                         image.fill(Qt::transparent);
+                         QPainter painter(&image);
+                         const QRectF& bbox = ScoreFont::fallbackFont()->bbox(id, 1);
+                         //default scale is 1.3, use smaller scale for large noteheads symbols
+                         qreal mag = std::min(1.3, std::min(w / bbox.width(), h / bbox.height()));
```
Comment (lasconic) [inline mscore/editdrumset.cpp]: Maybe it's my retina display but 2.6 looks a lot better on my mac. Same for master btw.
Maybe we should use *qApp->devicePixelRatio()

File: mscore/editdrumset.cpp [hunk 3]
```diff
@@ -102,14 +118,83 @@ EditDrumset::EditDrumset(const Drumset* ds, QWidget* parent)
       pitchList->setColumnWidth(1, 60);
       pitchList->setColumnWidth(2, 30);
 
+      QStringList validNoteheadRanges = { "Noteheads", "Slash noteheads", "Round and square noteheads", "Shape note noteheads", "Shape note noteheads supplement" };
```
Comment (lasconic) [inline mscore/editdrumset.cpp]: "Shape note noteheads supplement" doesn't exist in 2.3

File: mscore/editdrumset.cpp [hunk 4]
```diff
@@ -144,15 +232,46 @@ void EditDrumset::updateList2()
...
+void EditDrumset::setEnabledPitchControls(bool enable)
+      {
+      customGbox->setEnabled(enable);
+      noteHead->setEnabled(enable);
+      voice->setEnabled(enable);
+      shortcut->setEnabled(enable);
+      staffLine->setEnabled(enable);
+      stemDirection->setEnabled(enable);
+      drumNote->setEnabled(enable);
+      label_2->setEnabled(enable);
```
Comment (lasconic) [inline mscore/editdrumset.cpp]: worth giving some meaningful names no ?

File: mscore/editdrumset.cpp (hunk 2 above)
Comment (anatoly-os) [reply mscore/editdrumset.cpp]: Thanks, will experiment with it.

**Pull request merged**